src/App.vue

```diff
@@ -1,7 +1,13 @@
 <script setup lang="ts">
 import { RouterView } from 'vue-router'
+import NavBar from '@/components/NavBar.vue'
+import Footer from '@/components/Footer.vue'
 </script>
 
 <template>
-  <RouterView />
+  <NavBar :key="$route.fullPath" />
```

Comment:
Just curious, why is it necessary to add the key here?

Reply:
Ohhhh, is it because there are multiple navbars, each for different page?

Reply:
The key forces the navbar to reload each time you navigate to a new page which resets the sidebar